CorsixTH/Lua/app.lua

```diff
@@ -374,6 +377,25 @@ function App:init()
   return true
 end
 
+--! Works out the intended location of the gamelog file.
+--!return full path gamelog should exist at
+function App:getGamelogPath()
+  local config_path = self.command_line["config-file"] or ""
+  config_path = config_path:match("^(.-)[^" .. pathsep .. "]*$")
+  return config_path .. "gamelog.txt"
+end
+
+--! Checks and creates the gamelog file if it does not exist.
+function App:initGamelogFile()
+  local gamelog_path = self:getGamelogPath()
+  local gamelog = io.open(gamelog_path, "r")
+  if gamelog then return gamelog:close() end
```

Comment:
Why do you `return gamelog:close()` - what is that value?

It's unused in this patch. I would have expected just: `gamelog:close()`

Reply:
It was
`if gamelog and gamelog:close() then return end`. Then it was `if gamelong then return gamelog:close end`.
Maybe I'm trying to save space where I shouldn't.

Reply:
That's logically a different statement too. Revert to what it was please.